src/js/components/Calendar/Calendar.js

```diff
@@ -596,15 +576,49 @@ const Calendar = forwardRef(
 
         setDates(nextDates);
         if (date && typeof date === 'string') {
-          // for internal calculations, adjust based on timestamp
-          // and user's local timezone
-          setDate(normalizeForTimezone(nextDate, timestamp));
+          setDate(nextDate);
         } else if (date && Array.isArray(date)) {
           setDate(nextDates);
         }
         setActive(new Date(selectedDate));
         if (onSelect) {
+          // output date with no timestamp if that's how user provided it
```

Comment:
Can we move all of this conversion logic into a separate utility function? I think that would make this callback code a bit easier to follow.

Reply:
I separated this out into its own function. Additionally, I was able to remove blocks of this logic that were never being entered (confirmed via manual testing and coverage analysis in jest).